### DISCUSSION TITLE
Technical Lead Team rolling issues, shoutouts, and meeting deciders - Part 1, discussed (mostly) at TLT Meeting 8 Nov 2022

### DISCUSSION THREAD

**Discussion by valeriupredoi** [ESMValGroup/Community#56] [Meetings]
Hey @ESMValGroup/technical-lead-development-team here is the rolling discussion item containing all that needs to be pointed out, so that if there are a number of issues to be resolved pronto we organize a meeting; also the place we advertize the periodic, non-special edition calls, that I will strive to organize monthly - to the very least not, every 2-3 months for sure :beer: @bouweandela pls go ahead and post us the stuff we talked about yesterday at the workshop (ESMValTool workshop 17-20 October 2022)

**Comment by bouweandela** [comment]
Release produdure
----------------

Should a feature freeze for the release mean that we allow only bugfixes after that, or will any pull request that is merged after the feature freeze but before the release be included in the release?

It looks like there are different opinions on this (see e.g. https://github.com/ESMValGroup/ESMValTool/pull/2734#discussion_r986870398), so it would be good to talk about this at the upcoming tech lead team meeting.

**Comment by bouweandela** [comment]
Another topic: should iris-esmf-regrid be added as a dependency and can it replace the regridding schemes that are currently implemented in ESMValCore?

**Reply by valeriupredoi** [reply]
**Discussed at the TLT Call 8 Nov 2022**

@zklaus mentions we added it as a pure dependency just for others to use it in diagnostics; @bouweandela mentions that Core bits can call it and use it, but one needs to install it manually, but @sloosvel mentions that some bits work with the regular iris regridder but crash with the iris-esmf one; also @schlunma mentions some schemes are missing from either (not complementary). @zklaus mentions it's encouraging to have the option for users to just call it and use it.The outlook is to include it in Core (with tests and performance etc)

**Reply by sloosvel** [reply]
I don't mind picking that up. I guess it would be a matter of adding it as a dependency? And then, for data that does not work with it, open the corresponding issues?

**Comment by bouweandela** [comment]
Administration of the ESMValTool VM would be a good topic for discussion. At the moment everyone needs root permissions on it to get anything done. It would be good to clean the permissions so only two people need root access.

**Reply by valeriupredoi** [reply]
**Discussed at the TLT Call 8 Nov 2022**

@bouweandela proposes two VMAdmins - possibly Bouwe and @remi-kazeroni to have everything done in terms of permissions, we agreed Bouwe and Remi as VMAdmins

**Reply by valeriupredoi** [reply]
also discussed: we should look at what we store on the VM and nuke what we don't need from release results, plus look at individual recipes that may produce large intermediary files that we atore

**Reply by valeriupredoi** [reply]
@remi-kazeroni and @bouweandela will have a look together :heart:

**Comment by valeriupredoi** [comment]
release procedure for ESMValTool is involving and not well documented - I'd like to discuss https://github.com/ESMValGroup/ESMValTool/issues/2895 and also the spread of release workload - I think we can safely say that 2 people (release managers) would be needed in the eventuality that RTW doesn't come online before the next release (even so, it might be experimental, and a full run of all the recipes might still be needed one last release)

**Reply by valeriupredoi** [reply]
I am thinking we are starting to pile up discussion points, and we have just made the 2.7.0 release, so I reckon we should have us a Techy call next week, @ESMValGroup/technical-lead-development-team please react with :+1: to this post if you agree :beer:

**Reply by valeriupredoi** [reply]
arite - seems we ought to get us a call next week then, lemme hatch us a Doodle

**Comment by valeriupredoi** [comment]
would also want to touch upon https://github.com/ESMValGroup/ESMValTool/issues/2895 (also mentioned in my previous comment) **and specifically** data dirs on The Big Levante

**Reply by bouweandela** [reply]
This would also be a good and related discussion topic: https://github.com/ESMValGroup/ESMValTool/pull/2883

**Comment by bouweandela** [comment]
another topic for discussion: what to do with the commit message when merging a pull request? We have some guidance here: https://docs.esmvaltool.org/en/latest/community/review.html#merge, but a quick look at `git log` shows that many of us do not follow this.

@ESMValGroup/esmvaltool-coreteam I just changed the settings to automatically populate the commit message with the pull request description, please comment if you (dis)like this!

**Reply by valeriupredoi** [reply]
**Discussed at the TLT Call 8 Nov 2022**

Commit messages should be nicer and more descriptive: compress and expand as needed - at merge point, right before merging, edit the text in the merge text box

**Comment by zklaus** [comment]
Discord or alternatives should be discussed as well. It seems Discord has some difficulties for us, so we should affirm its use or decide to try an alternative, such as Slack or going back to Gitter/Matrix.

**Reply by bouweandela** [reply]
Yes, discord has been a bit of a struggle for me so far. First, my account was disabled immediately after creation. Then I wrote their support and they enabled it. But when I tried to log in today I got the message that my account has been disabled again. This isn't really the experience we would hope for for our users.

There is also the issue of the Met Office IT blocking the discord website.

I'm not keen on going back to Gitter because it doesn't offer any audio/video options. Slack could be an option, but it looks like you have to pay for it if you want to be able to use video calls with more than 2 people..

**Reply by katjaweigel** [reply]
For slack you also have to pay, if you want to keep messages for more than 90 days (Another project I know is discussing to leave slack for that reason at the moment and switch to mattermost, but I dont think, that has video?)

**Reply by zklaus** [reply]
According to [their pricing information](https://mattermost.com/pricing/), the free tier offers 1:1 calls, but no group calls.

Disclaimer: At SMHI we use self-hosted Mattermost as the company's instant messenger. I am happy with it.

**Reply by valeriupredoi** [reply]
RE Discord - seems like they even stopped writing back - I am currently in a loop of finding live plants in pots in those anti-bot tests, then promptly being told I am too dodgy, then immediately back to plants in pots. A real joke :man_facepalming:

**Reply by valeriupredoi** [reply]
**Discussed at the TLT Call 8 Nov 2022**

Discord it don't work for @bouweandela and @valeriupredoi - like really not; it works partially for @zklaus (missed messages). Klaus: a side clieant for Gitter - but maybe Mattermost vs Slack - Slack better but 90 day archive deleted. Freebie Slack set up by @zklaus - tey it out first among us (TLT) then toot about it

**Comment by bouweandela** [comment]
We also need a plan for maintaining the Zenodo records. Preferably we would need a shared email address, because it is not possible to have more than 1 email address associated with a Zenodo record.

**Reply by valeriupredoi** [reply]
**Discussed at the TLT Call 8 Nov 2022**

@bouweandela says we need a TLT email address where we can send/receive emails from for the Zenodo integration. @schlunma says there a Gmail delegated account (for groups)? We look into it - @valeriupredoi will look into this :+1:

**Reply by remi-kazeroni** [reply]
Support for multiple curators per community on Zenodo is considered but this might only be a long-term solution for us (see https://github.com/zenodo/zenodo/issues/810)

**Reply by bouweandela** [reply]
Looks like that is an open issue since 2015 (https://github.com/zenodo/zenodo/issues/35), so it may take a while.

**Comment by valeriupredoi** [comment]
I see issues are mounting. Shall we consider a 2h long call?

**Reply by valeriupredoi** [reply]
OK let me rephrase - I'm aiming for an 1h slot with the possibility of going overtime (like we usually do anyway) - so choose carefully when I send over the Doodler

**Comment by valeriupredoi** [comment]
Heads up @ESMValGroup/technical-lead-development-team we have a doodle for next week's call; you should have received an email about it, if not here's the link - https://doodle.com/meeting/participate/id/dL7v0Jgb - please fill out your preferred dates/times by this Friday 4 Nov, pub o'clock :beer: Consider this before you chose your pill, I mean time https://github.com/ESMValGroup/Community/discussions/56#discussioncomment-4028844 - we may have to spend a bit more than an hour, given we've got quite a few things on our plate (we don't have to go through them all though)
